test/extended/util/framework.go

```diff
@@ -938,6 +940,34 @@ func WaitForServiceAccount(c corev1client.ServiceAccountInterface, name string)
 	return wait.Poll(time.Duration(100*time.Millisecond), 3*time.Minute, waitFn)
 }
 
+// WaitForNamespaceSCCAnnotations waits until the cluster-policy-controller pods in the openshift-kube-controller-manager
+// namespace have added the SCC related annotations to the provided namespace;  NOTE: per
+// https://github.com/openshift/cluster-policy-controller/blob/master/pkg/security/controller/namespace_scc_allocation_controller.go#L74
+// the relist interval of that controller is 10 minutes, so are wait time here works off of that
+func WaitForNamespaceSCCAnnotations(c projectv1typedclient.ProjectV1Interface, name string) error {
+	waitFn := func() (bool, error) {
+		proj, err := c.Projects().Get(context.TODO(), name, metav1.GetOptions{})
+		if err != nil {
+			// it is assumed the project was created prior to calling this, so we
+			// do not distinguish not found errors
+			return false, err
+		}
+		if proj.Annotations == nil {
+			return false, nil
+		}
+		for k := range proj.Annotations {
+			// annotations to check based off of
+			// https://github.com/openshift/cluster-policy-controller/blob/master/pkg/security/controller/namespace_scc_allocation_controller.go#L112
+			if k == securityv1.UIDRangeAnnotation {
+				return true, nil
+			}
+		}
+		e2e.Logf("project %s current annotation set: %#v", name, proj.Annotations)
+		return false, nil
+	}
+	return wait.Poll(time.Duration(1*time.Minute), 11*time.Minute, waitFn)
```

Comment:
/cc @smarterclayton @deads2k 

11 minutes feels like way too much time to wait for a namespace to be fully ready. Any customer running a script that auto-deploys templates (as we do in the image-eco suite) will run into the namespace SCC issue.

Reply:
Yeah my thinking on basing on the relist interval is probably too cautious.  In assume the normal retry would fail enough times in a row for the update to get dropped.

Based on what I've seen in my test runs, 2 minutes should be more than enough, and is in line with what the other test framework `Wait...` methods are doing.

thanks